### PR TITLE
SUS-1737: In Wall and Comments tests double-check that user is logged in

### PR DESCRIPTION
**Pull request by TK-999**
https://wikia-inc.atlassian.net/browse/SUS-1737

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/commentstests/BlogCommentsTests.java [hunk 1]
```diff
@@ -3,29 +3,54 @@
...
+    blogPage = userProfile.openFirstPost();
+
+    if (user != User.ANONYMOUS) {
+      userName = user.getUserName();
+
+      // SUS-1737: In BlogComments tests sometimes user is not authenticated properly
+      // Let's check for this, and if this happens, log the situation and log in again
+      try {
+        blogPage.verifyUserLoggedIn(user);
+      } catch (TimeoutException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/commentstests/BlogCommentsTests.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/messagewall/MessageWallTests.java [hunk 2]
```diff
@@ -1,64 +1,83 @@
 package com.wikia.webdriver.testcases.messagewall;
 
+import com.wikia.webdriver.common.core.TestContext;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/messagewall/MessageWallTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.TestContext'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/messagewall/MessageWallTests.java [hunk 3]
```diff
@@ -1,64 +1,83 @@
...
+    User user = method.getAnnotation(Execute.class).asUser();
+
+    userName = user.getUserName();
+    wall = new MessageWall(driver).open(userName);
+
+    // SUS-1737: In Wall tests sometimes user is not authenticated properly
+    // Let's check for this, and if this happens, log the situation and log in again
+    try {
+      wall.verifyUserLoggedIn(user);
+    } catch (TimeoutException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/messagewall/MessageWallTests.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/testcases/commentstests/ArticleCommentsTests.java [hunk 4]
```diff
@@ -4,34 +4,59 @@
...
+
+    if (user != User.ANONYMOUS) {
+      userName = user.getUserName();
 
-  Credentials credentials = Configuration.getCredentials();
+      // SUS-1737: In Comments tests sometimes user is not authenticated properly
+      // Let's check for this, and if this happens, log the situation and log in again
+      try {
+        article.verifyUserLoggedIn(user);
+      } catch (TimeoutException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/commentstests/ArticleCommentsTests.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/common/core/helpers/User.java [hunk 5]
```diff
@@ -24,10 +24,12 @@
   REGULAR_USER_JAPAN("ci.user.language2.username", "ci.user.language2.password"),
   USER_GO_SEARCH_PREFERRED("ci.user.goSearchPreferredUser.username",
                            "ci.user.goSearchPreferredUser.password"),
+  // TODO: Is this correct?
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/helpers/User.java]: INFO Complete the task associated to this TODO comment. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 3 major
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/commentstests/BlogCommentsTests.java (hunk 1 above)
Comment (TK-999) [inline src/test/java/com/wikia/webdriver/testcases/commentstests/BlogCommentsTests.java]: Exception will be rethrown if login attempt fails.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 3 major
* INFO 1 info

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java [hunk 6]
```diff
@@ -303,20 +304,28 @@ public void afterNavigateTo(String url, WebDriver driver) {
...
-      if (user != null && user != User.ANONYMOUS) {
+    if (method.isAnnotationPresent(Execute.class)) {
+      user = method.getAnnotation(Execute.class).asUser();
+    }
+
+    if (user != User.ANONYMOUS) {
+      WikiBasePageObject basePageObject = new WikiBasePageObject();
+      try {
+        basePageObject.verifyUserLoggedIn(user);
+      } catch (TimeoutException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**